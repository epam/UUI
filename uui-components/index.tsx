/* Required so far, as uui-build assumes .index.tsx as an entry. We'll switch to ./src/index.tsx later */
export * from './src/index';
export * from './src/i18n';
