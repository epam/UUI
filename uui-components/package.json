{
  "name": "@epam/uui-components",
  "version": "4.2.0",
  "description": "EPAM UUI Components",
  "author": "EPAM",
  "license": "MIT",
  "main": "index.js",
  "typings": "index.d.ts",
  "scripts": {
    "build": "node ../uui-build/scripts/buildModule",
    "prepublish": "yarn build"
  },
  "dependencies": {
    "@epam/uui": "4.2.0",
    "@popperjs/core": "2.9.0",
    "@types/classnames": "2.2.6",
    "@types/lodash.clone": "4.5.6",
    "@types/lodash.isequal": "4.5.5",
    "@types/lodash.orderby": "4.6.6",
    "@types/lodash.range": "3.2.6",
    "@types/query-string": "6.1.0",
    "@types/react-custom-scrollbars": "4.0.3",
    "@types/react-measure": "2.0.6",
    "@types/react-transition-group": "4.4.1",
    "@types/resize-observer-browser": "0.1.5",
    "classnames": "2.2.6",
    "csstype": "2.6.10",
    "dayjs": "^1.10.6",
    "lodash.clone": "4.5.0",
    "lodash.isequal": "4.5.0",
    "lodash.orderby": "4.6.0",
    "lodash.range": "3.2.0",
    "react-custom-scrollbars-2": "^4.4.0",
    "react-focus-lock": "2.5.2",
    "react-measure": "2.5.2",
    "react-popper": "2.2.4",
    "react-transition-group": "4.4.2"
  },
  "peerDependencies": {
    "prop-types": "*",
    "react": ">=16.0.0",
    "react-dom": ">=16.0.0"
  },
  "devDependencies": {
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.10.0"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/epam/UUI"
  },
  "gitHead": "a46206079e94ed85069ae90a59cc4ba48b9bc9eb",
  "publishConfig": {
    "directory": "build"
  }
}
