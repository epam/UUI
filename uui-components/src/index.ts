import '../assets/styles/index.scss';
export * from './Text';
export * from './layout';
export * from './overlays';
export * from './dnd';
export * from './inputs';
export * from './buttons';
export * from './widgets';
export * from './table';
export * from './navigation';
export * from './RichTextView';
export * from './pickers';
export * from './errors';
export * from './helpers';
