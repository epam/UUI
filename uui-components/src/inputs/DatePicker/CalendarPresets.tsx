import * as React from "react";
import { IHasCX, cx, IHasRawProps, IHasForwardedRef, RangeDatePickerPresets, RangeDatePickerPresetValue } from "@epam/uui-core";

export const uuiPresets = {
    container: "uui-presets-container",
    header: "uui-presets-header",
    item: "uui-presets-item",
} as const;

export interface CalendarPresetsProps extends IHasCX, IHasRawProps<React.HTMLAttributes<HTMLDivElement>>, IHasForwardedRef<HTMLDivElement> {
    presets: RangeDatePickerPresets;
    onPresetSet: (nV: RangeDatePickerPresetValue) => void;
}

export class CalendarPresets extends React.Component<CalendarPresetsProps> {
    getPresets() {
        const presets = [];

        for (const key in this.props.presets) {
            if (this.props.presets[key]) {
                presets.push({
                    ...this.props.presets[key].getRange(),
                    name: this.props.presets[key].name,
                });
            }
        }

        return presets.sort((a, b) => a.order - b.order);
    }

    render() {
        return (
            <div ref={ this.props.forwardedRef } className={ cx(uuiPresets.container, this.props.cx) } { ...this.props.rawProps }>
                <div className={ uuiPresets.header }>Presets</div>
                { this.getPresets().map(item => (
                    <div
                        key={ item.name }
                        className={ uuiPresets.item }
                        onClick={ () => this.props.onPresetSet(item) }
                    >
                        { item.name }
                    </div>
                )) }
            </div>
        );
    }
}
