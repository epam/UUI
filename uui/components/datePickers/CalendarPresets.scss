@use  '../../assets/styles/index' as *;

.root {
    :global(.uui-presets-container) {
        padding: 24px;
        font-family: var(--font-regular);
    }

    :global(.uui-presets-header) {
        font-family: var(--font-semibold);
        font-size: 14px;
        line-height: 24px;
        padding-bottom: 12px;
    }

    :global(.uui-presets-item) {
        font-family: var(--font-semibold);
        font-size: 12px;
        line-height: 18px;
        color: var(--uui-link-button-text);
        padding-bottom: 6px;

        &:hover {
            color: var(--uui-link-button-text-active);
            cursor: pointer;
        }
    }
}
