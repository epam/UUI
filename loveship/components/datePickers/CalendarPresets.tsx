import css from './CalendarPresets.scss';
import { CalendarPresets as uuiCalendarPresets, CalendarPresetsProps } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';

export function applyCalendarPresetsMods() {
    return [css.root];
}

export const CalendarPresets = withMods<CalendarPresetsProps>(uuiCalendarPresets, applyCalendarPresetsMods, () => ({}));
