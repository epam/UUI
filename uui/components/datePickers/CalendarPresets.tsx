import css from './CalendarPresets.module.scss';
import { CalendarPresets as uuiCalendarPresets, CalendarPresetsProps } from '@epam/uui-components'; // TODO: rework presets to use LinkButton
import { withMods } from '@epam/uui-core';

export function applyCalendarPresetsMods() {
    return [css.root];
}

export const CalendarPresets = /* @__PURE__ */withMods<CalendarPresetsProps>(uuiCalendarPresets, applyCalendarPresetsMods, () => ({}));
