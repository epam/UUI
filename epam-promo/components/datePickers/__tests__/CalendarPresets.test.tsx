import React from 'react';
import { CalendarPresets } from '../CalendarPresets';
import renderer from 'react-test-renderer';
import { rangeDatePickerPresets } from '@epam/uui-components';

describe('CalendarPresets', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<CalendarPresets presets={rangeDatePickerPresets} onPresetSet={jest.fn} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
