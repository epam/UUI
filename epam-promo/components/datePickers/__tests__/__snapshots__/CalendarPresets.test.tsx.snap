// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarPresets should be rendered correctly 1`] = `
<div
  className="uui-presets-container root"
>
  <div
    className="uui-presets-header"
  >
    Presets
  </div>
  <div
    className="uui-presets-item"
    onClick={[Function]}
  >
    Today
  </div>
  <div
    className="uui-presets-item"
    onClick={[Function]}
  >
    This Week
  </div>
  <div
    className="uui-presets-item"
    onClick={[Function]}
  >
    Last Week
  </div>
  <div
    className="uui-presets-item"
    onClick={[Function]}
  >
    This Month
  </div>
  <div
    className="uui-presets-item"
    onClick={[Function]}
  >
    Last Month
  </div>
  <div
    className="uui-presets-item"
    onClick={[Function]}
  >
    Last 3 Months
  </div>
  <div
    className="uui-presets-item"
    onClick={[Function]}
  >
    This Year
  </div>
  <div
    className="uui-presets-item"
    onClick={[Function]}
  >
    Last Year
  </div>
</div>
`;
